omnigibson/object_states/attached_to.py

```diff
@@ -0,0 +1,293 @@
+from omni.physx.bindings._physx import ContactEventType
+
+import numpy as np
+from collections import defaultdict
+
+import omnigibson as og
+from omnigibson.macros import create_module_macros
+import omnigibson.utils.transform_utils as T
+from omnigibson.object_states.contact_subscribed_state_mixin import ContactSubscribedStateMixin
+from omnigibson.object_states.joint_break_subscribed_state_mixin import JointBreakSubscribedStateMixin
+from omnigibson.object_states.update_state_mixin import UpdateStateMixin
+from omnigibson.object_states.object_state_base import BooleanState, RelativeObjectState
+from omnigibson.object_states.link_based_state_mixin import LinkBasedStateMixin
+from omnigibson.object_states.contact_bodies import ContactBodies
+from omnigibson.utils.constants import JointType
+from omnigibson.utils.usd_utils import create_joint
+from omnigibson.utils.ui_utils import create_module_logger, suppress_omni_log
+from omnigibson.utils.python_utils import classproperty
+
+# Create module logger
+log = create_module_logger(module_name=__name__)
+
+
+# Create settings for this module
+m = create_module_macros(module_path=__file__)
+
+m.ATTACHMENT_LINK_PREFIX = "attachment"
+
+m.DEFAULT_POSITION_THRESHOLD = 0.05  # 5cm
+m.DEFAULT_ORIENTATION_THRESHOLD = np.deg2rad(5.0)  # 5 degrees
+m.DEFAULT_JOINT_TYPE = JointType.JOINT_FIXED
+m.DEFAULT_BREAK_FORCE = 10000  # Newton
+m.DEFAULT_BREAK_TORQUE = 10000  # Newton-Meter
+
+
+class AttachedTo(RelativeObjectState, BooleanState, ContactSubscribedStateMixin, JointBreakSubscribedStateMixin, LinkBasedStateMixin):
+    """
+        Handles attachment between two rigid objects, by creating a fixed/spherical joint between self.obj (child) and
+        other (parent). At any given moment, an object can only be attached to at most one other object, i.e.
+        a parent can have multiple children, but a child can only have one parent. Note that only
+        child.states[AttachedTo].get_value(parent) will return True.
```

Comment:
Just to clarify, parents == female, and children == male? Is this always true? We should maybe be explicit here

Reply:
done